lang/c++/api/NodeConcepts.hh

```diff
@@ -169,51 +157,47 @@ struct MultiAttribute
         return attrs_.at(index);
     }
 
-  private:
+private:
 
     std::vector<Attribute> attrs_;
 };
 
-
 template<typename T>
 struct NameIndexConcept {
 
     bool lookup(const std::string &name, size_t &index) const {
         throw Exception("Name index does not exist");
-        return 0;
     }
 
-    bool add(const::std::string &name, size_t index) {
+    bool add(const ::std::string &name, size_t) {
```

Comment:
`add(const ::std::string &name, size_t /*index*/)` seems more readable.  It also appears name is unused.

Reply:
Will fix.